src/v/cloud_storage/remote_partition.cc

```diff
@@ -627,9 +627,13 @@ remote_partition::get_term_last_offset(model::term_id term) const {
         }
     }
     // if last segment term is equal to the one we look for return it
-    auto last = _manifest.rbegin()->second;
-    if (last.segment_term == term) {
-        return last.next_kafka_offset() - kafka::offset(1);
+    auto last = _manifest.last_segment();
+    vassert(
+      last.has_value(),
+      "The manifest for {} is not expected to be empty",
+      _manifest.get_ntp());
+    if (last->segment_term == term) {
+        return last->next_kafka_offset();
```

Comment:
this method returns exclusive offset which belongs to the next segment

Reply:
thanks, this came from the cherry-pick where this had been committed_kafka_offset(), and I changed it back to next_kafka_offset without reinstating the subtraction of 1.

Fixed.